src/metabase/models/card.clj

```diff
@@ -510,7 +510,7 @@
       ;; TODO: should be done in after-update
       ;; has to place it here because changes is not available in after-update hook see toucan2#129
       (when (contains? changes :dataset_query)
-        (query-analysis/analyze-async! id))
+        (query-analysis/analyze-async! changes))
```

Comment:
if you need the `after-update` state of the card you can use the `card` argument of `t2/define-before-update`.
This way the card that goes into `analyze-async!` will have all the fields, including `archived`

Reply:
I'm pretty happy with this as is - there's nothing dangerous about analyzing an inactive card, and its very likely that we'll update the query of a card while its in active. Will try out the change you suggest and see which I prefer.

Reply:
Hmm, I prefer leaving as is, as that hook is currently a functional chain and I don't like muddying that with a side-effect.